beacon/goclient/proposer.go

```diff
@@ -230,9 +231,11 @@ func (gc *goClient) createValidatorRegistration(pubkey []byte, feeRecipient bell
 	return signedReg
 }
 
-func (gc *goClient) registrationSubmitter(slots <-chan phase0.Slot) {
-	for currentSlot := range slots {
-		gc.submitRegistrationsFromCache(currentSlot)
+func (gc *goClient) startSlotTicker(slotTickerProvider slotticker.Provider) {
+	ticker := slotTickerProvider()
+	for {
+		<-ticker.Next()
+		gc.submitRegistrationsFromCache(ticker.Slot())
 	}
```

Comment:
I would also check for `ctx.Done()` here using select. since we have ctx in `goClient`. this is only for node shutown, but still I'm not a fan of never-ending loops.